packages/astro/src/runtime/server/render/util.ts

```diff
@@ -28,6 +28,7 @@ const kebab = (k: string) =>
 	k.toLowerCase() === k ? k : k.replace(/[A-Z]/g, (match) => `-${match.toLowerCase()}`);
 const toStyleString = (obj: Record<string, any>) =>
 	Object.entries(obj)
+		.filter(([k, v]) => v !== undefined && v !== null)
```

Comment:
I'm not familiar with JSX, does it ignore all falsy values or just nullish? (and is there a reason to accept falsy values?)

Reply:
I can't find its source code, so I did a little more [tests](https://reactplayground.vercel.app/#N4IglgdgJgpgHgOgFYGcQC5wFsAOB7AJwBcACAQRxxIDMC8sSByBAegp2RTkYB0IxchUgCUYAQwDGpWvSYFxU3v0HESwEhPliiMYXjykAvjToNGWqQFoo9FhIA2YGBCJK+EvBBSkPLsZBgCEgBeEhsJAFcsZyIEAHMYIgBRexholwAhAE8ASSgACnN9VwBKd09vEjoDEI0tHT0DfN8ifwhAsohq2PloQPy+EhJRSVjNcR0UtJiBiCGhkakEAGUiAjApAFk8WAAaQfmICPt7fbn5xbH6mCn0onz2TqHOkpBdkHZOOAwQPgF8VSXEyycwKVzlLykdi1fIlEIAPjUB3kRAiBDms3mJAAPAALACMJG8WVSwWAwAOWJYLBoniIyzAAC8YOgSAAGSnzDz2QisxhMM5Yobc3nswVCkUEVlrCIwcVYyWs6hiewoOWc4V4HlSpgkMQAIwkjHl831kgA1nE6BFoABhLWi-kkLIwE54ADuAo1JEc7QAEjAwHFcUQ+QajZzDIZ4QG3diWAT4QdOoY+Gn2nAAaRYMrjlDKCBDEA).

It seems to have more complicate validation rules than just nullish or falsy values.
```jsx
<h1 style={{
      // falsy value `0` is preserved in fontSize
      // fontSize: 0
     // color omits any value that is not a `color`
      color: ' ',
      color: 0,
      color: true,
      color: false,
      color: ' abc',
      backgroundColor: '  yellow ',
      lineHeight: 'abc'
    }}>Hello</h1>
```